backend/src/model/dto.rs

```diff
@@ -63,10 +63,15 @@ pub struct PlantsSummaryDto {
     pub common_name: Option<Vec<Option<String>>>,
 }
 
-#[allow(clippy::missing_docs_in_private_items)] // TODO: See #97.
+/// Query parameters for the searching plants.
 #[typeshare]
 #[derive(Debug, Deserialize)]
 pub struct PlantsSearchParameters {
+    /// The system will check if this string occurs in the plants common name or binomial name.
```

Comment:
I have thought a bit about this over the past few days and came to the conclusion that we probably shouldn't search by German and English common names at the same time.

As a user, I would personally find it very confusing if I started typing "fi" and suddenly spruces (or "Fichten" in German) showed up in the search results, when I actually wanted to find firs.
In my opinion it would probably be better to add some mechanism to select which type of common name should be searched for. Preferably depending on the selected language.

If you want us to implement the search this way none the less we should at least show the German common name beside the English common name.

Reply:
But, whathever we choose to do should certainly be part of its own PR to avoid the same mistake we made in the last PR for the seed use case....

Reply:
I definitely agree about not putting everything in one PR, I think our feedback cycle is fast enough that this is not a problem. Probably I can already merge already now, or is something missing here?

The thing about German/English is that sometimes I remember the German name, sometimes the English. Some plants are quite exotic, the German name might be quite unknown/unusual. E.g. for [Pawpaw](https://de.wikipedia.org/wiki/Dreilappige_Papau) in principle has has the German names Indianerbanane or Papau but many people in Austria (including Yvonne and me) call them Pawpaw. Of course we could add "Pawpaw" also to German common names but this would be extra effort in maintaining the plants database.

Furthermore, I thought that a solution where we search everything at once makes the functionality (or at least the API) easier. Is this assumption wrong?

> As a user, I would personally find it very confusing if I started typing "fi" and suddenly spruces (or "Fichten" in German) showed up in the search results, when I actually wanted to find firs.

What if we simply rank English names in English settings higher? (When we implement the ranking.) Full matches also should be very high in ranking, so when typing "fir", "fir" definitely should be the first match, see [update of use case](https://github.com/ElektraInitiative/PermaplanT/pull/251/files#diff-ffcf3684203226512e824186cca963094610c83f56ecdffd18393acf5c281600)

Reply:
> The thing about German/English is that sometimes I remember the German name, sometimes the English. Some plants are quite exotic, the German name might be quite unknown/unusual. E.g. for [Pawpaw](https://de.wikipedia.org/wiki/Dreilappige_Papau) in principle has has the German names Indianerbanane or Papau but many people in Austria (including Yvonne and me) call them Pawpaw. Of course we could add "Pawpaw" also to German common names but this would be extra effort in maintaining the plants database.

Fair point. 
However: shouldn't "Pawpaw" then also be considered a German common name, since its frequently used to refer to a that specific plant even though its a word of foreign origin?   
"Physalis" is also commonly used to refer to a ["Blasenkirsche"](https://de.wikipedia.org/wiki/Blasenkirschen) for example.
Before finding the linked Wikipedia article I wasn't even aware that it had a "proper" German name to be honest.

And even if you can't remember the name of a plant in your preferred language, is it too much to ask of the user to either change the interface language or do 30 second lookup in an online dictionary or wikipedia?

Again, I don't want to sound like I'm trying to avoid work here but this idea is again something that

a. Might be more complicated to implement than expected.
b. Potentially has a very negative impact on user experience. 
c. Solves a rather specific problem that can easily be overcome using external tools.

This is why I'm of the opinion that the risks are not worth the rather small reward.

> Furthermore, I thought that a solution where we search everything at once makes the functionality (or at least the API) easier. Is this assumption wrong?

Yes, I think so.
If we go with the ranking method you described we would still have to communicate the users preferred language to the backend so that we know how the results should be ranked. 
This is exactly the same information needed to determine which column should be used by the search if we limit go with the "to one language at a time" approach.

Reply:
> However: shouldn't "Pawpaw" then also be considered a German common name anyway, since its frequently used to refer to a that specific plant even though its a word of foreign origin?

As said, it is simply a huge effort to do that. It is better to improve the search than to duplicate many entries in the database.

> Still, if you can't remember the name of a plant in your preferred language, is it too much to ask of the user to either change the interface language or do 30 second lookup in an online dictionary or wikipedia?

This doesn't help you, as you would need to exactly know what is in **our database under which language**. When we are not able to present a match, it would lead to frustration.

> a. Might be more complicated to implement than expected, delaying the first milestone even further.

If there is something specifically which is complicated, please tell.

> b. Potentially has a very negative impact on user experience.

Presenting no match definitely is the worst user experience. This is also why search machines nowadays more or less ignore what you type and present you matches anyway.

> c. Solves a rather specific problem that can easily be overcome using external tools.

As said: unfortunately this is **not** the case. Yes, you could look up the German name ["Blasenkirsche](https://de.wikipedia.org/wiki/Blasenkirschen), but what if we **don't** have this name but have Andenkirsche instead? Or Kapstachelbeere? There is no external tool telling you what a person needs to type in our search field so that she will get results.

I updated #251, I hope it is fine now.

Reply:
Ok, before I say anything else: I will go ahead and implement this in another PR as outlined in the new usecase over the course of the next week.

Just to explain my reasoning a bit:
> As said, it is simply a huge effort to do that.

Absolutely! I'm not arguing against that.  I meant to say that if *any kind of term* is commonly used to refer to a plant in German, it should *by definition* already be in the database as a common name in that language. If that isn't the case in the majority of cases we should probably reevaluate where we are getting our data from.

> If there is something specifically which is complicated, please tell.

The thing is, I have never worked on a search feature before (except maybe a few trivial ones), let alone in rust with actix web.
So its nothing specific really.
However, getting the ranking right could be a bit tricky (E.g. preventing elements from appearing twice in the search and so on).

> Presenting no match definitely is the worst user experience.

Maybe. Generally I would agree, but I think we all know about the "Kaufhaus Österreich" debacle where random stuff appeared in the product search. That played a big part in making the site borderline unusable.

Although I'd say this could be easily prevented in our case by making it obvious why a certain item was included in the search results. E.g. by highlighting the search term.

> As said: unfortunately this is not the case.

Again, debatable. For example: I don't think many non-German English speakers would benefit from search results coming from the German common name.  In the end it all comes down to who our user base is and what they expect.

> Yes, you could look up the German name ["Blasenkirsche](https://de.wikipedia.org/wiki/Blasenkirschen), but what if we don't have this name but have Andenkirsche instead? Or Kapstachelbeere? There is no external tool telling you what a person needs to type in our search field so that she will get results.

If all else fails you should hopefully be able to find the Latin name of a plant using the common name you know best.

Reply:
Turned it into its own issue.
https://github.com/ElektraInitiative/PermaplanT/issues/258